### PR TITLE
New Resource & Data Source: `azurerm_snapshot`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
- New Resource: `azurerm_snapshot`
- New Data Source: `azurerm_snapshot`
- Re-organising all the Compute resources into a `Compute` section in the sidebar

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Resource tests pass:

screen shot 2017-10-12 at 14 27 25

**Comment by tombuildsstuff** [comment]
Snapshot Data Source tests pass:

screen shot 2017-10-12 at 16 04 18

File: website/docs/r/snapshot.html.markdown [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+* `name` - (Required) Specifies the name of the Snapshot resource. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the resource group in which to create the Snapshot. Changing this forces a new resource to be created.
+
+* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
+
+* `create_option` - (Required) Indicates how the snapshot is to be created. Possible values are `Copy` or `Import`. Changing this forces a new resource to be created.
+
+~> **Note:** One of `source_uri`, `source_resource_id` or `storage_account_id` must be specified.
```
Comment (mbfrahry) [inline website/docs/r/snapshot.html.markdown]: Do these conflict with each other or can you have all of them specified?

**Review by mbfrahry — APPROVED** [review]
LGTM! Just a question

File: website/docs/r/snapshot.html.markdown (hunk 1 above)
Comment (tombuildsstuff) [reply website/docs/r/snapshot.html.markdown]: I assumed we'd be able to make these conflicting fields - but apparently you can have them all specified ¯\_(ツ)_/¯

Comment (mbfrahry) [reply website/docs/r/snapshot.html.markdown]: That's weird but alright let's ship it!

**Pull request merged**